Documentation/ApiOverview/Enumerations/Enumeration.rst

```diff
@@ -0,0 +1,134 @@
+
+.. include:: ../../Includes.txt
+
+.. _Enumerations-How-to-use:
+
+=======================
+How To Use Enumerations
+=======================
+
+
+.. _Enumerations-Create-an-Enumeration:
+
+Create an Enumeration
+=====================
+
+To create a new enumeration you have to extend the class :php:`TYPO3\CMS\Core\Type\Enumeration`,
+make sure your enumeration is marked :php:`final`, this ensures your code only receives a known
+set of values. Otherwise adding more values by extension will lead to undefined behavior in your code.
+
+Values are defined as constants in your implementation. The names of the constants must be given in uppercase.
+
+A special, optional constant :php:`__default` represents the default value of your enumeration
+if it is present. In that case the enumeration can be instantiated without a value and
+will be set to the default.
+
+Example::
+
+   final class LikeWildcard extends \TYPO3\CMS\Core\Type\Enumeration
+   {
+      const __default = self::BOTH;
+
+      /** @var int Do not use any wildcard */
+      const NONE = 0;
+
+      /** @var int Use wildcard on left side */
+      const LEFT = 1;
+
+      /** @var int Use wildcard on right side */
+      const RIGHT = 2;
+
+      /** @var int Use wildcard on both sides */
+      const BOTH = 3;
+   }
+
+
+.. _Enumerations-Use-an-Enumeration:
+
+Use an Enumeration
+==================
+
+You always *have to* use the :php:`Enumeration::cast()` method
+for instantiation.
```

Comment:
I know it's just moving content but this is not true. Using `new` is perfectly fine too and guarantees that the resulting value is of the desired type. The static `cast()` method cannot ensure this (PHP 8.0 will have a `static` return type for this)